### ISSUE TITLE
QoL - Option to lock perk rolls to return to a previous mountain to roll.

### ISSUE THREAD

**Issue by Teslan-1** [TwoAbove/noita-tools#88]
I am on a run where going to the third mountain for perks is best to collect 'Perk Lottery' and return to a previous mountain and roll to the next.

The Main issue is if I roll on the Third machine, then return to roll the Second machine, the app will roll the third forward.

potential idea could be to have a button, or tick box to lock the mountain as is.
![Screenshot 2022-01-13 170153](https://user-images.githubusercontent.com/97710435/149427960-b12e4428-f5ba-45f3-882a-00de822868d6.png)
![Screenshot 2022-01-13 170316](https://user-images.githubusercontent.com/97710435/149428033-2d9a5dbb-b357-423c-b1e2-12d5e6981b28.png)

The issue lies in Gamble, where actual perks given would be Fungal Disease and Fire, Like this:
![image](https://user-images.githubusercontent.com/97710435/149428157-1c18cd92-3a78-4505-aa08-edb7b7710e81.png)

Thanks, for taking a look! Love the tool!

**Comment by TwoAbove** [comment]
Hey! Interesting edge case that I didn't think of.

I'll need some time to think about how to solve this, since the holy mountain perks get re-calculated after each change. 

I see why thins can be useful, so I think I'll need to refactor how the perks and changes are stored and calculated

**Comment by TwoAbove** [comment]
Thanks for this!

**Comment by Teslan-1** [comment]
I am not very code savvy, but if I can think of any way to explain something that may work I'll let you know.
For now all I can think of is a way to 'lock' specific perks using a toggle.

For example, I select the perks I like, hit lock. that perk set should no longer be change-able. So if I lock 'Perk set 3' at 2 rolls unlocked mountains should still get affected if I were to roll a different mountain.

Lock toggle could be renamed as a 'Collected' or similar name signifying there are no perks left.
Or if you wanted to get really fancy the Selected Perks could act as that, Locking you from rolling those perks.

I have also noticed in actual play if I took perk 1 and 2, and rolled I would get the next sequential perk in the re-roll stack rather than the one associated with the same slot. This could be considered another issue on its own but I feel it has its roots in this issue.

I feel like I did a lot of rambling, but I hope it helps at least a little. :)
